pkg/providers/instance/instance.go

```diff
@@ -193,6 +197,19 @@ func (p *Provider) createAKSIdentifyingExtension(ctx context.Context, vmName str
 	return nil
 }
 
+func (p *Provider) createCSExtension(ctx context.Context, vmName string, cse string, isWindows bool) (err error) {
+	vmExt := p.getCSExtension(cse, isWindows)
+	vmExtName := *vmExt.Name
+	logging.FromContext(ctx).Debugf("Creating virtual machine CSE for %s", vmName)
+	v, err := createVirtualMachineExtension(ctx, p.AZClient.virtualMachinesExtensionClient, p.resourceGroup, vmName, vmExtName, *vmExt)
+	if err != nil {
+		logging.FromContext(ctx).Errorf("Creating VM CSE for VM %q failed, %w", vmName, err)
+		return fmt.Errorf("creating VM CSE for VM %q, %w failed", vmName, err)
+	}
+	logging.FromContext(ctx).Debugf("Created virtual machine CSE for %s, with an id of %s", vmName, *v.ID)
+	return nil
+}
```

Comment:
Wondering if there is a cleaner way to share the vm extension creation logic between the different extensions. Thoughts?